FirebaseTestProject/Views/PostsList.swift

```diff
@@ -8,25 +8,96 @@
 import SwiftUI
 
 struct PostsList: View {
-    @StateObject var postData = PostData()
+    @StateObject var postData: PostData
+    @State private var isPresenting: Bool = false
+    @State private var isEditing = false
+    @State private var searchText: String = ""
+    @Namespace var namespace
     
     var body: some View {
         NavigationView {
-            List(postData.posts, id: \.text) { post in
-                PostRow(post: post)
-            }
-            .navigationTitle("Posts")
-            .onAppear {
-                Task {
-                    await postData.loadPosts()
+            VStack {
+                // Searchable Posts
+                SearchBar(text: $searchText)
+                let posts = searchText == "" ?
+                                    postData.posts :
+                                    postData.posts.filter({ $0.postSearch(searchText) })
+                
+                let prefSize = 0.9
+                ScrollView(.vertical) {
+                    GeometryReader { g in
```

Comment:
Is the `GeometryReader` needed here?  We can use it, but we'd need an article for it.

Reply:
LMK: It was the ONLY way I found to resize a view within a parent view to make things look neater.